gnomad_qc/v4/resources/basics.py

```diff
@@ -54,19 +54,37 @@ def get_gnomad_v4_vds(
     # Count current number of samples in the VDS
     n_samples = vds.variant_data.count_cols()
 
-    # Obtain withdrawn UKBB samples (includes 5 samples that should be removed from the VDS)
-    excluded_ukbb_samples_ht = hl.import_table(
-        ukbb_excluded_samples_path,
-        no_header=True,
-    ).key_by("f0")
+    # Obtain samples in application 31063 withdraw list
+    ukbb_31063_withdrawn_ht = hl.import_table(ukbb_excluded_31063_path, no_header=True)
+    ukbb_31063_withdrawn_ht = ukbb_31063_withdrawn_ht.key_by(
+        eid_31063=ukbb_31063_withdrawn_ht.f0
+    )
+
+    # Obtain samples in application 26041 withdraw list
+    ukbb_26041_withdrawn_ht = hl.import_table(ukbb_excluded_26041_path, no_header=True)
+    ukbb_26041_withdrawn_ht = ukbb_26041_withdrawn_ht.key_by(
+        eid_26041=ukbb_26041_withdrawn_ht.f0
+    )
+
+    ukbb_application_map_ht = ukbb_application_map.ht()
+    withdrawn_ukbb_samples = ukbb_application_map_ht.filter(
+        hl.literal(ukbb_31063_withdrawn_ht.eid_31063.collect()).contains(
+            ukbb_application_map_ht.eid_31063
+        )
+        | hl.literal(ukbb_26041_withdrawn_ht.eid_26041.collect()).contains(
+            ukbb_application_map_ht.eid_26041
+        )
+    ).s.collect()
 
-    sample_map_ht = ukbb_array_sample_map.ht()
+    sample_map_ht = ukbb_array_sample_map.ht().key_by("ukbb_app_26041_id")
     sample_map_ht = sample_map_ht.annotate(
         withdrawn_consent=hl.is_defined(
-            excluded_ukbb_samples_ht[sample_map_ht.ukbb_app_26041_id]
+            ukbb_26041_withdrawn_ht[sample_map_ht.ukbb_app_26041_id]
```

Comment:
I think we can use 31063 withdrawn here right? You will need the map you loaded above

Reply:
There's one sample in the 26041 withdraw list but not in  the 31061 one
```
withdrawn_26041 = ukb_excluded.versions['26041_20210809'].ht().eid_26041.collect()
withdrawn_31063 = ukb_excluded.versions['31063_20220222'].ht().eid_31063.collect()
mapping_ht_ukb_withdrawn_from_31063 = set(mapping_ht_ukb.filter(hl.literal(withdrawn_31063).contains(mapping_ht_ukb.eid_31063)).s.collect())
mapping_ht_ukb_withdrawn_from_26041 = set(mapping_ht_ukb.filter(hl.literal(withdrawn_26041).contains(mapping_ht_ukb.eid_26041)).s.collect())
print("Number withdrawn samples in both 26041 and 31063: " + str(len(mapping_ht_ukb_withdrawn_from_31063 & mapping_ht_ukb_withdrawn_from_26041)),
      "\nNumber withdrawn samples in only 31063: " + str(len(mapping_ht_ukb_withdrawn_from_31063 - mapping_ht_ukb_withdrawn_from_26041)),
      "\nNumber withdrawn samples in only 26041: " + str(len(mapping_ht_ukb_withdrawn_from_26041 - mapping_ht_ukb_withdrawn_from_31063)),
)     
```

Number withdrawn samples in both 26041 and 31063: 27 
Number withdrawn samples in only 31063: 45 
Number withdrawn samples in only 26041: 1